Source/Fuse.Marshal/Marshal.Cast.uno

```diff
@@ -142,9 +142,25 @@ namespace Fuse
 			@return true if converted successfully, false if no suitable conversion exists. `null` cannot be converted and will return false;
 			@param value the result value (0 padded as necessary)
 			@param size the size of the result
+			
+			@throw None This function will not throw any exceptions.
 		*/
 		public static bool TryToZeroFloat4(object o, out float4 value, out int size)
 		{
+ 			try
+ 			{
+				return TryToZeroFloat4Impl(o, out value, out size);
+ 			}
+ 			catch (Exception ex)
+ 			{
+ 				value = float4(0);
+ 				size = 0;
+ 				return false;
+ 			}
+		}
+		
+		static bool TryToZeroFloat4Impl(object o, out float4 value, out int size)
+		{
```

Comment:
I thought the same way at first, and had just blocked FromHex from throwing. But then I noticed other places can also cause exceptions (in the test case, plus possibly more). The solution I have is the only one that didn't involve rewriting a lot of the functionality here. Even if we do cleanup a lot of this stuff there is still no guarantee -- as far as I know C#/Uno doesn't have a no-throw clause for a function.

Reply:
Still, it would be much better if we could catch around those other places, so it's possible to hunt them down one by one

By the way, this inspired me to [send this PR](https://github.com/fusetools/uno/pull/1383) to fix the `Uno.Color.FromHex`-issue. So at least one of these is going to be easier in the near future.

Reply:
I also want to stress that I think converting *every* `Exception` to a "silent" error is very bad; out-of-memory errors and bad method arguments (null-pointers etc) deep down in other call-trees will be treated as "we can't marshal this", which is often plainly the wrong thing to do.